packages/framework-core/src/decorators/entity.ts

```diff
@@ -1,22 +1,60 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
 import { Booster } from '../booster'
-import { Class, EntityInterface, ReducerMetadata, EventInterface } from '@boostercloud/framework-types'
+import {
+  Class,
+  EntityInterface,
+  ReducerMetadata,
+  EventInterface,
+  RoleAccess,
+  AnyClass,
+} from '@boostercloud/framework-types'
 import 'reflect-metadata'
+
+interface EntityAttributes {
+  authorizeReadEvents: RoleAccess['authorize']
+}
+
+type EntityDecoratorParam = AnyClass | EntityAttributes
+type EntityDecoratorResult<TEntity, TParam> = TParam extends EntityAttributes
+  ? (entityClass: Class<TEntity>) => void
+  : void
+
 /**
  * Decorator to register a class as an Entity
  * @constructor
  */
-export function Entity<TEntity extends EntityInterface>(entityClass: Class<TEntity>): void {
-  Booster.configureCurrentEnv((config): void => {
-    if (config.entities[entityClass.name]) {
-      throw new Error(`An entity called ${entityClass.name} is already registered
+export function Entity<TEntity extends EntityInterface, TParam extends EntityDecoratorParam>(
+  classOrAttributes: TParam
+): EntityDecoratorResult<TEntity, TParam> {
+  // The craziness with the param and return types is to achieve that either:
+  // - The @Entity decorator doesn't have parenthesis: THEN we need to accept a class as a parameter and return void
+  // - The @Entity decorator have parenthesis: THEN we need to accept an object with attributes and return a function accepting a class as a parameter
```

Comment:
Check this comment. This way, if you don't want to expose an entity, you can just write `@Entity` and not `@Entity()`